controllers/core/node_controller.go

```diff
@@ -60,13 +61,16 @@ func (r *NodeReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
-	err := r.Manager.AddOrUpdateNode(node)
-	if err != nil {
-		logger.Error(err, "failed to add node")
-		return ctrl.Result{}, err
+	_, found := r.Manager.GetNode(req.Name)
```

Comment:
Do any tests exercise these changes in `node_controller` ?

Reply:
This was missing, I added Unit tests for this.